### PR TITLE
Bug 1588096 - only support scope expressions, not array notation

### PR DESCRIPTION
**Pull request by djmitche**
This replaces a few uses of the old `[[scope]]` notation for scope
requirements with scope expressions, and removes support for that
notation.

Bugzilla Bug: [1588096](https://bugzilla.mozilla.org/show_bug.cgi?id=1588096)

### CODE REVIEW

**Comment by djmitche** [comment]
Please have a careful look at the changes to auth and web-server.  Those changes continue to pass tests, but please verify the logic.

File: libraries/scopes/src/satisfaction.js [hunk 1]
```diff
@@ -1,13 +1,4 @@
...
-const {validateScopeSets} = require('./validate');
-
-const validateScopePatterns = (scopePatterns) => {
-  assert(scopePatterns instanceof Array && scopePatterns.every((scope) => {
-    return typeof scope === 'string';
-  }), 'scopes must be an array of strings');
-};
-
-const patternMatch = exports.patternMatch = (pattern, scope) => {
+exports.patternMatch = (pattern, scope) => {
```
Comment (imbstack) [inline libraries/scopes/src/satisfaction.js]: should we just rename this file at this point if it only has `patternMatch` in it?

File: services/auth/src/signaturevalidator.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ const limitClientWithExt = function(credentialName, issuingClientId, accessToken
     }
 
     // Validate certificate scopes are subset of client
-    if (!utils.scopeMatch(scopes, [cert.scopes])) {
+    if (!utils.satisfiesExpression(scopes, {AllOf: cert.scopes})) {
```
Comment (imbstack) [inline services/auth/src/signaturevalidator.js]: just a side note that I really like how much clearer the new syntax is to read

**Comment by imbstack** [comment]
Also needs a changelog :)

File: libraries/scopes/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 Simple utilities to validate scopes, scope-sets, and scope-expression satisfiability.
 
-For information on scopes, see [the Taskcluster documentation](../../ui/docs/manual/design/apis/hawk/scopes.mdx).
+For information on scopes, see [the Taskcluster documentation](https://docs.taskcluster.net/docs/manual/design/apis/hawk/scopes).
```
[the remark this answers is not in the document]
Comment (djmitche) [reply libraries/scopes/README.md]: I figure link to the highest-quality stable URL.  This README is read on GitHub, so we can't use a link to the "current deployment", but `docs.taskcluster.net` is stable.  The old link directly to the .mdx file was not high-quality :)

File: services/auth/src/signaturevalidator.js (hunk 2 above)
Comment (djmitche) [reply services/auth/src/signaturevalidator.js]: Definitely, it was a good improvement..

File: libraries/scopes/src/satisfaction.js (hunk 1 above)
Comment (djmitche) [reply libraries/scopes/src/satisfaction.js]: Yeah, honestly most of the code could fit in one file at this point..

File: services/auth/src/api.js [hunk 4]
```diff
@@ -844,7 +844,7 @@ builder.declare({
 
       // Check scopes
       const formerRoleScopes = this.resolver.resolve(role.scopes);
-      const scopesAdded = input.scopes.filter(s => !scopeUtils.scopeMatch(formerRoleScopes, [[s]]));
+      const scopesAdded = input.scopes.filter(s => !scopeUtils.satisfiesExpression(formerRoleScopes, s));
```
Comment (owlishDeveloper) [inline services/auth/src/api.js]: just to make sure: so we are now sure to expect the new syntax..... is it an array? Not an object like `{AllOf: []}` or `{AnyOf: []}` etc?

Comment (djmitche) [reply services/auth/src/api.js]: In this case, `s` is a single scope (a string).

Comment (owlishDeveloper) [reply services/auth/src/api.js]: sorry, I meant `input.scopes`

Comment (owlishDeveloper) [reply services/auth/src/api.js]: Basically, I just wanted to double check that we don't loop with `Array.prototype.filter` over an object. If this worry is unfounded, r+

Comment (djmitche) [reply services/auth/src/api.js]: Good question -- yeah, that's enforced with JSON-schema.

**Pull request merged**